src/logohunter.py

```diff
@@ -0,0 +1,349 @@
+#
+# MASTER SCRIPT TO DETECT LOGOS IN IMAGE AND FIND MATCHES
+# for each image find and pass logos, compute features, see if logo passes cutoff, save image
+#
+import matplotlib.pyplot as plt
+
+import sys, os
+from PIL import Image
+import cv2
+import numpy as np
+
+import argparse
+
+from keras_yolo3.yolo import YOLO
+from utils import contents_of_bbox, load_features, bbox_colors, parse_input, chain_preprocess, load_extractor_model
+from similarity import features_from_image, similarity_cutoff, load_brands_compute_cutoffs, similar_matches, draw_matches
+
+from timeit import default_timer as timer
+
+input_shape = (299,299,3)
+sim_threshold = 0.95
+
+
+def detect_logo(yolo, img_path, save_img, save_img_path='./', postfix=''):
+    """
+    Call YOLO logo detector on input image, optionally save resulting image.
+
+    Args:
+      yolo: keras-yolo3 initialized YOLO instance
+      img_path: path to image file
+      save_img: bool to save annotated image
+      save_img_path: path to directory where to save image
+      postfix: string to add to filenames
```

Comment:
also add the variables returned, pep8 guidelines: https://www.python.org/dev/peps/pep-0008/#global-variable-names

Reply:
Thanks, I hadn't noticed!